roles/openshift_metering/tasks/install.yaml

```diff
@@ -0,0 +1,93 @@
+---
+- name: Create temp directory for doing work in on target
+  command: mktemp -td openshift-metering-ansible-XXXXXX
+  register: mktemp
+  changed_when: False
+
+- name: Ensures {{ mktemp.stdout }}/metering-manifests dir exists
+  file:
+    path: "{{ mktemp.stdout }}/metering-manifests"
+    state: directory
+
+- name: Copy files to temp directory
+  copy:
+    src: "{{ item }}"
+    dest: "{{ mktemp.stdout }}/metering-manifests/{{ item | basename }}"
+  with_fileglob:
+  - "*.yaml"
+
+- name: Copy admin client config
+  command: >
+    cp {{ openshift.common.config_base }}/master/admin.kubeconfig {{ mktemp.stdout }}/admin.kubeconfig
+  changed_when: false
+
+- name: Update Metering Helm Operator deployment image
+  yedit:
+    src: "{{ mktemp.stdout }}/metering-manifests/metering-helm-operator-deployment.yaml"
+    edits:
+    - key: spec.template.spec.containers[0].image
+      value: "{{ openshift_metering_operator_image }}"
+    - key: spec.template.spec.containers[1].image
+      value: "{{ openshift_metering_operator_image }}"
+  when: openshift_metering_operator_image != ""
+
+- name: Update Metering config
+  yedit:
+    src: "{{ mktemp.stdout }}/metering-manifests/metering.yaml"
+    edits:
+    - key: spec
+      value: "{{ openshift_metering_config }}"
+  when: openshift_metering_config != None
+
+- name: "Create {{ openshift_metering_namespace }} project"
+  oc_project:
+    state: present
+    name: "{{ openshift_metering_namespace }}"
+    description: Openshift Metering
+    node_selector: ""
+
+- name: Create metering namespace viewer cluster role
+  oc_clusterrole:
+    state: present
+    name: openshift-metering-namespace-viewer
+    rules:
+    - apiGroups: [""]
+      resources: ["namespaces"]
+      verbs: ["get"]
+
+- name: Grant access to openshift-metering with namespace viewer role
+  shell: >
```

Comment:
Is there a reason why you're using shell and adding `|| true` to the end?

Reply:
The `oc_adm_policy_user` module doesn't allow specifying the name of the `ClusterRoleBinding` created, it re-uses the `ClusterRole`'s name. I want to be able to create a unique name so that if we were to run this playbook in a cluster with two different namespaces, the ClusterRoleBinding (which is global) it won't conflict with an existing one by using the unique name. The `|| true` is to handle the case where it already exists, in the case of re-running, then it should already be correctly set.

This made me realize I need to handle this ClusterRoleBinding uninstall however.

Reply:
There is an option to update the `failed_when` condition on the task so that even if it already exists it won't fail out. 
In other places we've done a check prior to creating so that we only execute the task if it doesn't already exist.

I think one of those two would be preferable, in case the creation of the clusterrolebinding failed for another reason

Reply:
Gotcha, Ill try that.